### PR TITLE
Adding docker images with the name of the branch

### LINKED ISSUE #36: Create docker images for PRs with labels
When putting a pull-request out of draft, create all docker images with the label of the branch.

### PR DESCRIPTION
**Pull request by ineiti**
Closes #36

Thanks go ChatGPT for the DockerTag environment...

Testing by adding a draft pull request first.
16h53: marked as ready for review

### CODE REVIEW

File: .github/workflows/build-docker.yml [hunk 1]
```diff
@@ -4,11 +4,17 @@ on:
...
     branches:
       - main
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
 
 jobs:
   build-docker:
     name: Build D-Voting Docker images
     runs-on: ubuntu-22.04
+    if: '! github.event.pull_request.draft'
```
Comment (PascalinDe) [inline .github/workflows/build-docker.yml:14]: I don't agree with that - I can very clearly see wanting to create the Docker image for a PR that's not yet ready for testing purposes

it's not merged anyway, I don't see the advantage of creating a third case - either it's on main, then it should have the latest tag or it is not, then it gets a "non-latest" tag anyway

Comment (ineiti) [reply .github/workflows/build-docker.yml:14]: I thought of the poor github servers and the environment: if you mark a PR as `draft`, then it's mostly to show that you're currently working on this feature/fix/whatever. All tests happen locally on your computer, you create the images locally, etc.

Only once you mark it as ready for review do your fellow engineers need to have access to the docker images.

Comment (PascalinDe) [reply .github/workflows/build-docker.yml:14]: OK - I mainly want to avoid overcomplicating this

Comment (ineiti) [reply .github/workflows/build-docker.yml:14]: If we get confused by it, we can still remove it later.

**Review by PascalinDe — APPROVED** [review]

**Pull request merged**